### ISSUE TITLE
Use absolute URLs in API section

### ISSUE THREAD

**Issue by mturoci** [h2oai/wave#1667]
We use relative links in our [API section](https://wave.h2o.ai/docs/api/) for submodules for example. However, these won't work if user comes directly to `/docs/api` (without trailing `/`). The proper solution here should be to use absolute links.

Repro:

Click [this link without trailing slash](https://wave.h2o.ai/docs/api) and try to navigate to a submodule - should result in 404.

**Comment by JacksonIsaac** [comment]
Working on this

**Comment by JacksonIsaac** [comment]
Looks like the docs/api files are no longer in the repo ? I was able to setup the docs on my local system and make the changes, but no way to track the file changes

**Comment by mturoci** [comment]
Hi @JacksonIsaac the resulting md files are generated using `pdoc3`, thus not part of the repo. Changes need to be made [here](https://github.com/h2oai/wave/blob/master/py/docs/templates/text.mako). You can test it out using `cd py && make docs` which will run the python docs build step.

**Comment by JacksonIsaac** [comment]
I tried testing by 'make docs', and the build directory does get created, but the sub-folders are empty. The log does show the files being listed but something is going wrong perhaps. Could you maybe guide me please on this ? :)

Here's my output log: https://privatebin.net/?89b64ef4945fa8a5#8pM5tqdzdm9HL7g7a58MnJQQ4uLWADEfcTVJ3UGSwXTD

**Comment by mturoci** [comment]
> build directory does get created, but the sub-folders are empty

Yep, this is expected. If you have a closer look at the [make target](https://github.com/h2oai/wave/blob/master/py/Makefile#L33-L34) you can notice that the built assets are then moved over to `website/docs/api` folder. That's where you can find the results.

Note: Once your fix is ready, you need to run docs website to verify everything works as expected.

**Comment by JacksonIsaac** [comment]
I tried few different approaches for this,

added target="_self" in <a> tags, or href="./{}", but both doesn't seem to lead me anywhere. Any more ideas perhaps ? :)

I am assuming it can be something on link(m) https://github.com/h2oai/wave/blob/master/py/docs/templates/text.mako#L94 that I can try, or trying to handle submodules, but not sure how this works

**Comment by mturoci** [comment]
Seems like this is the [function](https://github.com/h2oai/wave/blob/95a2674c1501a30d958eba0f1d384de0d60fb068/py/docs/templates/text.mako#L8) you need to look into.

**Comment by JacksonIsaac** [comment]
I have been researching a bit, and also tried few different ways to try solve this, without much luck.

Also, at pdocs for Doc.url:https://pdoc3.github.io/pdoc/doc/pdoc/#pdoc.Doc.url I see it is about relative URL. Couldn't find much about absolute URL (External object has a .url which returns absolute URL). Not sure if I am looking at the right place.

Do you perhaps any guidance on this ? :)

**Comment by mturoci** [comment]
I would start by looking into the correct docs. We use pdoc3, not pdoc :) I would first aim for some simple solution that pdoc3 would provide for free. If not successful, I would look at the code around and see if I can construct the absolute URLs manually which should basically be not so hard.
